hathor/version.py

```diff
@@ -12,4 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = '0.52.1'
+import os
+import re
+from typing import Optional
+
+from structlog import get_logger
+
+BASE_VERSION = '0.52.1'
+
+DEFAULT_VERSION_SUFFIX = "-local"
+BUILD_VERSION_FILE_PATH = "./BUILD_VERSION"
+
+# Valid formats: 1.2.3, 1.2.3-rc.1 and nightly-ab49c20f
+BUILD_VERSION_REGEX = r"^(\d+\.\d+\.\d+(-rc\.\d+)?|nightly-[a-f0-9]{7,8})$"
+
+
+logger = get_logger()
+
+
+def _get_build_version() -> Optional[str]:
+    """Try to get the build version from BUILD_VERSION_FILE_PATH and validate it.
+
+    :return: The build version or None, if there is no file or the version is invalid.
+    """
+    if not os.path.isfile(BUILD_VERSION_FILE_PATH):
+        return None
+
+    with open(BUILD_VERSION_FILE_PATH, 'r') as f:
+        build_version = f.readline()
+        match = re.match(BUILD_VERSION_REGEX, build_version)
+
+        if match:
+            return build_version
+        else:
+            logger.warn("A build version with an invalid format was found. Ignoring it.", build_version=build_version)
+            return None
+
+
+def _get_version() -> str:
+    """Get the current hathor-core version from the build version or the default one with a local suffix
+
+    :return: The current hathor-core version
+    """
+    local_version = BASE_VERSION + DEFAULT_VERSION_SUFFIX
```

Comment:
We need to incluide the commit hash here as well, as accorded in the thread https://github.com/HathorNetwork/ops-tools/pull/627#discussion_r1157673133